pkg/kv/export/utils.go

```diff
@@ -0,0 +1,71 @@
+package export
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"errors"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/treeverse/lakefs/pkg/logging"
+)
+
+var ErrIllegalFilePath = errors.New("illegal file path")
+
+// Untar takes a destination dir and a reader to a tar file.
+// Looping over the tar file, extracting files into dir
+func Untar(dir string, r io.Reader) error {
+	gzr, err := gzip.NewReader(r)
+	if err != nil {
+		return err
+	}
+	defer gzr.Close()
+
+	tr := tar.NewReader(gzr)
+
+	for {
+		header, err := tr.Next()
+		switch {
+		case err == io.EOF:
+			return nil
+		case err != nil:
+			return err
+		// Skip if the header is nil
+		case header == nil:
+			continue
+		}
+		path := filepath.Join(dir, header.Name) // nolint:gosec
+		if !strings.HasPrefix(path, filepath.Clean(dir)) {
+			logging.Default().WithError(ErrIllegalFilePath).Error("Illegal file path:", path)
```

Comment:
As this function is very basic I would keep the logging to the caller and just return more information on the error